src/shared/unit-file.c

```diff
@@ -21,3 +22,54 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_TIMER,
                       UNIT_PATH);
 }
+
+int unit_validate_symlink(const char *filename, const char *target) {
+        const char *src, *dst;
+        UnitType t, t2;
+        int r, r2;
+
+        src = basename(filename);
+        dst = basename(target);
+
+        /* Reject cases where the alias or the target unit have instance names. */
+        r = unit_name_to_instance(src, NULL);
```

Comment:
maybe add a wrapper like this:

```c
#define unit_name_classify(name) unit_name_to_instance(name, NULL);
```

Otherwise it's really hard to grok what the game played is